Can't compile with -Dgc_none since v1.4

Try to compile a Crystal program with `-Dgc_none` on Linux (e.g. an empty file), and you'll see the following linker error (or another `pthread` symbol):

```console
$ cat /dev/null > bug.cr
$ crystal build -Dgc_none bug.cr
/usr/bin/ld: T-hread5858M-utex.o: référence au symbole non défini « pthread_mutexattr_settype@@GLIBC_2.2.5 »
//lib/x86_64-linux-gnu/libpthread.so.0 : erreur lors de l'ajout de symboles : DSO manquant dans la ligne de commande
collect2: error: ld returned 1 exit status
Error: execution of command failed with code: 1: `cc "${@}" -o /home/julien/.cache/crystal/crystal-run-bug.tmp  -rdynamic -L/usr/bin/../lib/crystal -lpcre -lm -levent -lrt -ldl`
```

The linker command is indeed missing `-lpthread`. If I manually specify it, then it will compile:

```console
$ crystal run -Dgc_none --link-flags="-lpthread" bug.cr
```

The regression was introduced in Crystal 1.4. It used to work in previous releases.

--- a/src/lib_c/aarch64-linux-gnu/c/pthread.cr
+++ b/src/lib_c/aarch64-linux-gnu/c/pthread.cr
@@ -6,7 +6,7 @@ require "./sys/types"
 # static libraries. So we just skip `pthread` entirely in interpreted mode.
 # The symbols are still available in the interpreter because they are loaded in the compiler.
 {% unless flag?(:interpreted) %}
-@[Link("pthread")]
+  @[Link("pthread")]
 {% end %}
 lib LibC
   PTHREAD_MUTEX_ERRORCHECK = 2

--- a/src/lib_c/aarch64-linux-gnu/c/pthread.cr
+++ b/src/lib_c/aarch64-linux-gnu/c/pthread.cr
@@ -1,5 +1,13 @@
 require "./sys/types"
 
+# Starting with glibc 2.34, `pthread` is integrated into `libc` and may not even
+# be available as a separate shared library.
+# There's always a static library for compiled mode, but `Crystal::Loader` does not support
+# static libraries. So we just skip `pthread` entirely in interpreted mode.
+# The symbols are still available in the interpreter because they are loaded in the compiler.
+{% unless flag?(:interpreted) %}
+@[Link("pthread")]
+{% end %}
 lib LibC
   PTHREAD_MUTEX_ERRORCHECK = 2
 

--- a/src/lib_c/arm-linux-gnueabihf/c/pthread.cr
+++ b/src/lib_c/arm-linux-gnueabihf/c/pthread.cr
@@ -6,7 +6,7 @@ require "./sys/types"
 # static libraries. So we just skip `pthread` entirely in interpreted mode.
 # The symbols are still available in the interpreter because they are loaded in the compiler.
 {% unless flag?(:interpreted) %}
-@[Link("pthread")]
+  @[Link("pthread")]
 {% end %}
 lib LibC
   PTHREAD_MUTEX_ERRORCHECK = 2

--- a/src/lib_c/arm-linux-gnueabihf/c/pthread.cr
+++ b/src/lib_c/arm-linux-gnueabihf/c/pthread.cr
@@ -1,5 +1,13 @@
 require "./sys/types"
 
+# Starting with glibc 2.34, `pthread` is integrated into `libc` and may not even
+# be available as a separate shared library.
+# There's always a static library for compiled mode, but `Crystal::Loader` does not support
+# static libraries. So we just skip `pthread` entirely in interpreted mode.
+# The symbols are still available in the interpreter because they are loaded in the compiler.
+{% unless flag?(:interpreted) %}
+@[Link("pthread")]
+{% end %}
 lib LibC
   PTHREAD_MUTEX_ERRORCHECK = 2
 

--- a/src/lib_c/i386-linux-gnu/c/pthread.cr
+++ b/src/lib_c/i386-linux-gnu/c/pthread.cr
@@ -6,7 +6,7 @@ require "./sys/types"
 # static libraries. So we just skip `pthread` entirely in interpreted mode.
 # The symbols are still available in the interpreter because they are loaded in the compiler.
 {% unless flag?(:interpreted) %}
-@[Link("pthread")]
+  @[Link("pthread")]
 {% end %}
 lib LibC
   PTHREAD_MUTEX_ERRORCHECK = 2

--- a/src/lib_c/i386-linux-gnu/c/pthread.cr
+++ b/src/lib_c/i386-linux-gnu/c/pthread.cr
@@ -1,5 +1,13 @@
 require "./sys/types"
 
+# Starting with glibc 2.34, `pthread` is integrated into `libc` and may not even
+# be available as a separate shared library.
+# There's always a static library for compiled mode, but `Crystal::Loader` does not support
+# static libraries. So we just skip `pthread` entirely in interpreted mode.
+# The symbols are still available in the interpreter because they are loaded in the compiler.
+{% unless flag?(:interpreted) %}
+@[Link("pthread")]
+{% end %}
 lib LibC
   PTHREAD_MUTEX_ERRORCHECK = 2
 

--- a/src/lib_c/x86_64-linux-gnu/c/pthread.cr
+++ b/src/lib_c/x86_64-linux-gnu/c/pthread.cr
@@ -6,7 +6,7 @@ require "./sys/types"
 # static libraries. So we just skip `pthread` entirely in interpreted mode.
 # The symbols are still available in the interpreter because they are loaded in the compiler.
 {% unless flag?(:interpreted) %}
-@[Link("pthread")]
+  @[Link("pthread")]
 {% end %}
 lib LibC
   PTHREAD_MUTEX_ERRORCHECK = 2

--- a/src/lib_c/x86_64-linux-gnu/c/pthread.cr
+++ b/src/lib_c/x86_64-linux-gnu/c/pthread.cr
@@ -1,5 +1,13 @@
 require "./sys/types"
 
+# Starting with glibc 2.34, `pthread` is integrated into `libc` and may not even
+# be available as a separate shared library.
+# There's always a static library for compiled mode, but `Crystal::Loader` does not support
+# static libraries. So we just skip `pthread` entirely in interpreted mode.
+# The symbols are still available in the interpreter because they are loaded in the compiler.
+{% unless flag?(:interpreted) %}
+@[Link("pthread")]
+{% end %}
 lib LibC
   PTHREAD_MUTEX_ERRORCHECK = 2
 

